src/cli/test_command.zig

```diff
@@ -1432,12 +1457,36 @@ pub const TestCommand = struct {
                 (Environment.isWindows and (strings.startsWith(arg, ".\\") or
                     strings.startsWith(arg, "..\\")))) break true;
         } else false;
+
+        var line_filters = bun.StringHashMap(std.ArrayList(u32)).init(ctx.allocator);
+
         if (has_relative_path) {
             // One of the files is a filepath. Instead of treating the
             // arguments as filters, treat them as filepaths
             const file_or_dirnames = ctx.positionals[1..];
             for (file_or_dirnames) |arg| {
-                scanner.scan(arg) catch |err| switch (err) {
+                const file_line = parseFileLineArg(arg);
+                if (file_line) |parsed| {
+                    const abs_pattern = brk: {
+                        if (std.fs.path.isAbsolute(parsed.file_pattern)) {
+                            break :brk try ctx.allocator.dupe(u8, parsed.file_pattern);
+                        } else {
+                            var local_buf: bun.PathBuffer = undefined;
+                            const cwd = bun.fs.FileSystem.instance.top_level_dir;
+                            const joined = bun.path.joinAbsStringBuf(cwd, &local_buf, &.{parsed.file_pattern}, .auto);
+                            break :brk try ctx.allocator.dupe(u8, joined);
+                        }
+                    };
+
+                    const result = try line_filters.getOrPut(abs_pattern);
+                    if (!result.found_existing) {
+                        result.value_ptr.* = std.ArrayList(u32).init(ctx.allocator);
+                    }
+                    try result.value_ptr.append(parsed.line_num);
```

Comment:
The file paths likely need to be canonicalized. When the scanner finds the file path it matches, it should use the one it matches instead. This will better handle differences in trailing slashes or slash directions or filesystem casing. This will matter more for Windows than for POSIX.